backend/infrahub/core/schema/schema_branch.py

```
@@ -1231,6 +1232,79 @@ def validate_count_against_cardinality(self) -> None:
                             f"{node.kind}: Relationship {rel.name!r} max_count must be 0 or greater than 1 when cardinality is MANY"
                         )
 
+    def validate_inherited_relationships_fields(self) -> None:
+        for name in self.node_names:
+            node_schema = self.get(name=name, duplicate=False)
+            if not node_schema.inherit_from:
+                continue
+
+            self.validate_node_inherited_relationship_fields(node_schema)
+
+    def validate_node_inherited_relationship_fields(self, node_schema: NodeSchema) -> None:
```

something to think about is that we allow overriding inherited relationships and attributes
say, 2 generics define different versions of a `device` relationship and a node inherits from both of those generics AND overrides the `device` relationship. do we still want to raise a validation error in this case?
I'd say yes b/c it still seems incorrect for a node to inherit from generics with different definitions of the same field (and it is simpler), but just want to make sure it is considered

---

When we query each generic the relationship on each node inheriting from the generic **must** look the same. So depending on the override we have to do it, i.e. we can't change the type or the peer.